### PR TITLE
Add driver_colors function for easier plotting of lines from same team drivers

### PR DESCRIPTION
**Pull request by dialtone**
Pull request from this discussion https://github.com/theOehrly/Fast-F1/discussions/145

### CODE REVIEW

**Comment by theOehrly** [comment]
Hi, just so that you don't sit here wondering, I'll try to take a closer look at this today or tomorrow, hopefully.

File: fastf1/plotting.py [hunk 1]
```diff
@@ -138,6 +187,71 @@ def setup_mpl(mpl_timedelta_support=True, color_scheme='fastf1',
...
+        >>> driver_color('charles leclerc')
+        '#dc0000'
+        >>> driver_color('max verstappen')
+        '#0600ef'
+        >>> driver_color('ver')
+        '#0600ef'
+        >>> driver_color('lec')
+        '#dc0000'
+
+        shortened driver names, included sponsors and typos can be dealt with
```
Comment (theOehrly) [inline fastf1/plotting.py:212]: remove the sponsors part, not applicable here

File: fastf1/plotting.py [hunk 2]
```diff
@@ -138,6 +187,71 @@ def setup_mpl(mpl_timedelta_support=True, color_scheme='fastf1',
...
+        too (within reason)
+
+        >>> driver_color('Max Verst')
+        '#0600ef'
+        >>> driver_color('Charles')
+        '#dc0000'
+
+    Args:
+        identifier (str): Abbreviation or uniquely identifying name of the team.
```
Comment (theOehrly) [inline fastf1/plotting.py:221]: driver, not team; 
see comments in code below too

**Review by theOehrly — CHANGES_REQUESTED** [review]
Just some minor details. Also, please fix the lint failure.

**Comment by theOehrly** [comment]
Everything else looks good. I have also thought about the plain white Haas color again. We had discussed this shortly in #145. I'd say it's fine for now. Team colors are that way too and it works with FastF1s usual dark theme. Support for switching between light and dark theme should be added separately, and this should then include two separate color pallets. But this is a separate problem.

**Comment by theOehrly** [comment]
These two should be fixed so that ci passes cleanly

```
=================================== FAILURES ===================================
_________________________________ FLAKE8-check _________________________________
/home/runner/work/Fast-F1/Fast-F1/fastf1/plotting.py:190:1: E302 expected 2 blank lines, found 1
/home/runner/work/Fast-F1/Fast-F1/fastf1/plotting.py:256:1: E302 expected 2 blank lines, found 1
```

**Comment by dialtone** [comment]
I think I fixed the 2 CI errors.

**Review by theOehrly — APPROVED** [review]

**Comment by theOehrly** [comment]
Yes, looks all good now. Thanks a lot for contributing!

**Comment by dialtone** [comment]
my pleasure, thanks for the library in the first place

**Pull request merged**